pkg/cluster/cluster.go

```diff
@@ -259,6 +262,22 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				// TODO: we can't handle another upgrade while an upgrade is in progress
 				c.logger.Infof("spec update: from: %v to: %v", c.cluster.Spec, event.cluster.Spec)
 				c.cluster = event.cluster
+				staticSpec := c.cluster.Spec.ClusterTLS.Static
```

Comment:
let's only care TLS for the cluster creation phase initially? ignore the modification for now?